astropy/coordinates/angles.py

```diff
@@ -139,18 +141,25 @@ def __new__(cls, angle, unit=None, dtype=None, copy=True, **kwargs):
                                **kwargs)
 
     @staticmethod
-    def _tuple_to_float(angle, unit):
+    def _tuple_to_float(angle, unit, supress_deprecations=False):
         """
         Converts an angle represented as a 3-tuple or 2-tuple into a floating
         point number in the given unit.
         """
-        # TODO: Numpy array of tuples?
-        if unit == u.hourangle:
-            return form.hms_to_hours(*angle)
-        elif unit == u.degree:
-            return form.dms_to_degrees(*angle)
-        else:
-            raise u.UnitsError(f"Can not parse '{angle}' as unit '{unit}'")
+        with warnings.catch_warnings():
+            if supress_deprecations:
+                #TODO: eliminate this when the dms_to_degreess deprecation is no
+                # longer is needed - see
+                # https://github.com/astropy/astropy/issues/12239
+                # https://github.com/astropy/astropy/pull/13162
+                warnings.simplefilter('ignore', AstropyDeprecationWarning)
```

Comment:
I don't know how I feel about this. If we are serious about deprecating this, we should not even use it internally.

If we have no choice but to use it internally, then instead of deprecating, we should fix it properly.